File: leetcode/algorithms/bit-manipulation/reverse-integer.java
class Solution {
    public int reverse(int x) {
        
        Stack<Integer> iStk = new Stack<Integer>();
        int multiplier = x<0 ? -1 : 1;
        x = Math.abs(x);
        while(x>0){
            iStk.push(x%10);
            x = x/10;
        }
        
        int i=0;
        int reversedSum = 0;
        while(!iStk.isEmpty()){
            reversedSum+=(iStk.pop()*Math.pow(10,i));
            ++i;
        }
        
        return (reversedSum==Integer.MAX_VALUE ? 0 : multiplier*(reversedSum));
        
    }
}

File: leetcode/algorithms/math/reverse-integer.ts
const MAX_VALUE: number = Math.pow(2,31) - 1;
function reverse(x: number): number {
    const multiplier = x<0 ? -1 : 1;
    x*=multiplier;
    const stk: number[] = [];
    while(x>0){
        const curr = x%10;
        stk.push(curr);
        x = ~~(x/10);
    }
    
    
    let i: number = 0, sum: number = 0;
    while(stk.length>0){
        const curr: number = stk.pop();
        sum+=curr*Math.pow(10,i);
        ++i;
    }
    return sum>MAX_VALUE ? 0 : sum*multiplier;

};